### PR TITLE
Adjust stamina regain to 1s turns

### PR DESCRIPTION
**Pull request by nexusmrsep**
#### Summary

 ```SUMMARY: Balance "Adjust stamina regain to 1s turns"```

#### Purpose of change
Stamina regain was 6x faster after introduction of 1s turns.
This will bring the regain down to old value.

#### Describe the solution
Stamina regain is governed mainly by `player::update_stamina( int turns )`.
I divided the turns by 6 in the final calculation to return the ratio to the existing one.

#### Describe alternatives you've considered
There were theoretically few other ways to do it, since division by 6 seems a bit hacky:
* manipulating the calculation itself - too serious rework due to how calculation  revolves around low numbers
* multiplying the stamina pool by 6 - out of option since other places add and subtract stamina directly so it would break a lot of things.

#### Additional context
N/A

### CODE REVIEW

File: src/player.cpp [hunk 1]
```diff
@@ -4657,8 +4657,8 @@ void player::update_stamina( int turns )
             stamina_recovery += bonus;
         }
     }
-
-    stamina = roll_remainder( stamina + stamina_recovery * turns );
+    //  recover ratio was balanced for legacy 6s/turn, hence the division by 6
+    stamina = roll_remainder( stamina + stamina_recovery * turns / 6 );
```
Comment (KorGgenT) [inline src/player.cpp]:
```suggestion
    stamina = roll_remainder( stamina + stamina_recovery * turns / 6.0f );
```
I would do this just to make things very clear that this is not integer division.

File: src/player.cpp [hunk 2]
```diff
@@ -4657,8 +4657,8 @@ void player::update_stamina( int turns )
             stamina_recovery += bonus;
         }
     }
-
-    stamina = roll_remainder( stamina + stamina_recovery * turns );
+    //  recover ratio was balanced for legacy 6s/turn, hence the division by 6
+    stamina = roll_remainder( stamina + stamina_recovery * turns / 6.0f );
```
Comment (kevingranade) [inline src/player.cpp]: That int in there makes me nervous about truncation errors, I'm confident this version will not truncate.
```suggestion
    stamina += roll_remainder( stamina_recovery * turns / 6.0f );
```

**Comment by Leosky** [comment]
something got wrong. I loose stam walking bellow max carry weight, and i can't run across a street without needing 5minute regen

**Comment by kevingranade** [comment]
Need to revisit this, it does seem to have overshot.

**Pull request merged**